File: App-ListVeiculos/src/app/cliente/models/cliente.ts
import { Veiculo } from 'src/app/veiculo/models/veiculo';

export class Cliente {
    id!: string;
    nome!: string;
    telefone!: string;
    cpf!: string;
    dataNascimento!: number;
    veiculos!: Veiculo[]
}

File: ListaVeiculos/ListaVeiculos.Api/Models/Cliente.cs
using System.Collections.Generic;

namespace ListaVeiculos.Api.Models
{
    public class Cliente
    {
        public int Id { get; set; }
        public string Nome { get; set; }
        public string Telefone { get; set; }
        public string Cpf { get; set; }
        public string DataNascimento { get; set; }
        public IEnumerable<Veiculo> Veiculos { get; set; }

        public Cliente() { }

        public Cliente(int id, string nome, string telefone, string cpf, string dataNascimento)
        {
            Id = id;
            Nome = nome;
            Telefone = telefone;
            Cpf = cpf;
            DataNascimento = dataNascimento;
        }
    }
}
